docs/resources/data_source_config.md

```diff
@@ -0,0 +1,80 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "grafana_data_source_config Resource - terraform-provider-grafana"
+subcategory: "Grafana OSS"
+description: |-
+  Official documentation https://grafana.com/docs/grafana/latest/datasources/HTTP API https://grafana.com/docs/grafana/latest/developers/http_api/data_source/
+  The required arguments for this resource vary depending on the type of data
+  source selected (via the 'type' argument).
+---
+
+# grafana_data_source_config (Resource)
+
+* [Official documentation](https://grafana.com/docs/grafana/latest/datasources/)
+* [HTTP API](https://grafana.com/docs/grafana/latest/developers/http_api/data_source/)
+
+The required arguments for this resource vary depending on the type of data
+source selected (via the 'type' argument).
+
+Use this resource for configuring multiple Data Souces, when that configuration (`json_data_encoded` field) requires circular reference like in the example below.
+
+!> When using `grafana_data_source_config` resource, the corresponding `grafana_data_source` resources must contain `json_data_encoded` field ignored. Otherwise infinite update loop will occur.
+
+
+## Example Usage
+
+```terraform
+resource "grafana_data_source" "loki" {
+  type               = "loki"
+  name               = "loki"
+  url                = "http://localhost:3100"
+
+  lifecycle {
+    ignore_changes = [json_data_encoded]
+  }
+}
+
+resource "grafana_data_source" "tempo" {
+  type               = "tempo"
+  name               = "tempo"
+  url                = "http://localhost:3200"
+
+  lifecycle {
+    ignore_changes = [json_data_encoded]
+  }
+}
+
+resource "grafana_data_source_config" "loki" {
+  uid = grafana_data_source.loki.uid
+
+  json_data_encoded = jsonencode({
+    derivedFields = [
+      {
+        datasourceUid = grafana_data_source.tempo.uid
+        matcherRegex  = "[tT]race_?[iI][dD]\"?[:=]\"?(\\w+)"
+        matcherType   = "regex"
+        name          = "traceID"
+        url           = "$${__value.raw}"
+      }
+    ]
+  })
+}
+
+resource "grafana_data_source_config" "tempo" {
+  provider = grafana.http
+
+  uid = grafana_data_source.tempo.uid
+
+  json_data_encoded = jsonencode({
+    tracesToLogsV2 = {
+      customQuery     = true
+      datasourceUid   = grafana_data_source.loki.uid
+      filterBySpanID  = false
+      filterByTraceID = false
+      query           = "|=\"$${__trace.traceId}\" | json"
+    }
+  })
+}
```

Comment:
I'm going to assume that these docs were written manually because this example is not in a separate file. It should be. The example will also be tested automatically at that point.

The docs should be generated with `go generate ./...`